src/router/router.cpp

```diff
@@ -58,19 +62,31 @@ bool Router::is_valid_method(const std::string& method)
 std::string Router::build_request_path(const HttpRequest& request, const Location* best_match)
 {
     std::string full_path = config_.root;
+    std::cout << "config_.root = " << config_.root << std::endl;
     std::string folder = static_cast<std::string>(best_match->path);
     std::string file =
         request.path.substr(best_match->path.size(), request.path.size() - best_match->path.size());
```

Comment:
```suggestion
    // this will crash but the condition should never be true in the first place
    assert(best_match->path.size() <= request.path.size());
    std::string file =
        request.path.substr(best_match->path.size(), request.path.size() - best_match->path.size());
```

You substract unsigned values that could cause a nasty overflow (and they do when folder is size 0)